tests/test_astrometry.py

```diff
@@ -68,25 +69,50 @@ def t_rv():
 
 def test_ingest_parallaxes(temp_db, t_plx):
     # Test ingest of parallax data
-    ingest_parallaxes(
-        temp_db, t_plx["source"], t_plx["plx"], t_plx["plx_err"], t_plx["plx_ref"]
-    )
+
+    for row in t_plx:
+        ingest_parallax(
+            temp_db,
+            row["source"],
+            row["plx"],
+            row["plx_err"],
+            row["plx_ref"],
+        )
 
     results = (
         temp_db.query(temp_db.Parallaxes)
         .filter(temp_db.Parallaxes.c.reference == "Ref 1")
         .table()
     )
     assert len(results) == 2
+    assert not results["adopted"][0]  # 1st source with ref 1 should not be adopted
     results = (
         temp_db.query(temp_db.Parallaxes)
         .filter(temp_db.Parallaxes.c.reference == "Ref 2")
         .table()
     )
-    assert len(results) == 1
-    assert results["source"][0] == "Fake 3"
-    assert results["parallax"][0] == 155
-    assert results["parallax_error"][0] == 0.6
+    assert len(results) == 2
+    assert results["source"][1] == "Fake 3"
+    assert results["parallax"][1] == 155
+    assert results["parallax_error"][1] == 0.6
+    assert results["adopted"][0]  # 1st source with ref 2 should be adopted
+
+
+def test_parallax_exceptions(temp_db):
+    with pytest.raises(AstroDBError) as error_message:
+        ingest_parallax(temp_db, "bad source", 1, 1, "Ref 1")
+    assert "does not exist in Sources table" in str(error_message.value)
+
+    with pytest.raises(AstroDBError) as error_message:
+        ingest_parallax(temp_db, "Fake 1", 1, 1, "bad ref")
+    assert "does not exist in Publications table" in str(error_message.value)
+
+    ingest_parallax(temp_db, "Fake 2", 1, 1, "Ref 2")
+    with pytest.raises(AstroDBError) as error_message:
+        ingest_parallax(temp_db, "Fake 2", 1, 1, "Ref 2")
+    assert "Duplicate measurement exists with same reference" in str(
+        error_message.value
+    )
```

Comment:
Duplicate these tests with `raise_error = False`.  In each scenario, test all three elements of the flags dictionary.